client/dashboard/task-list/tasks/payments/payfast.js

```diff
@@ -0,0 +1,177 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component, Fragment } from '@wordpress/element';
+import { Button } from '@wordpress/components';
+import interpolateComponents from 'interpolate-components';
+import { compose } from '@wordpress/compose';
+import { withDispatch } from '@wordpress/data';
+
+/**
+ * WooCommerce dependencies
+ */
+import { ADMIN_URL } from '@woocommerce/wc-admin-settings';
+import { Form, Link, TextControl } from  '@woocommerce/components';
+
+/**
+ * Internal dependencies
+ */
+import { recordEvent } from 'lib/tracks';
+import withSelect from 'wc-api/with-select';
+
+class PayFast extends Component {
+  constructor( props ) {
+    super( props );
+  }
+
+  getInitialConfigValues = () => {
+    return {
+      merchant_id: '',
+      merchant_key: '',
+      pass_phrase: '',
+    };
+  };
+
+  validate = ( values ) => {
+    const errors = {};
+
+    if ( ! values.merchant_id ) {
+      errors.merchant_id = __( 'Please enter your merchant ID', 'woocommerce-admin' );
+    }
+
+    if ( ! values.merchant_key ) {
+      errors.merchant_key = __( 'Please enter your merchant key', 'woocommerce-admin' );
+    }
+
+    if ( ! values.pass_phrase ) {
+      errors.pass_phrase = __( 'Please enter your passphrase', 'woocommerce-admin' );
+    }
+
+    return errors;
+  };
+
+  updateSettings = async ( values ) => {
+    const {
+      createNotice,
+      isSettingsError,
+      updateOptions,
+      markConfigured,
+      setRequestPending,
+    } = this.props;
+
+    setRequestPending( true );
+
+    await updateOptions( {
```

Comment:
I received this notice since my store was still in USD.  We may want to go ahead and update the store currency (option name `woocommerce_currency`) to `ZAR` if they select this payment method.  May also be worth noting that this will happen in the initial payment method description.